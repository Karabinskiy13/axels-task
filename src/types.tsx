export type Tag = {
  id: string;
  text: string;
};

export type Image = {
  largeImageURL: string;
  id: number;
  previewURL: string;
  tags: string;
  hits?: [{ largeImageURL: string; id: number; previewURL: string; tags: string }];
};

export type Favorite = {
  id?: number;
  previewURL?: string;
};
