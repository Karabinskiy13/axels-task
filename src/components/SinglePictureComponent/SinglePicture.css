.picture{
display: flex;
flex-direction: column;
justify-content: center;
}
.picture__image{
    max-height: 150px;
}