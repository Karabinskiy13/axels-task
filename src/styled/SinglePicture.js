import styled from 'styled-components';
import { Card, Col } from 'react-bootstrap';

export const Picture = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
export const Columns = styled(Col)`
  margin-top: 20px;
  cursor: pointer;
`;
export const Modal = styled.button`
  border: none;
  background: none;
`;
