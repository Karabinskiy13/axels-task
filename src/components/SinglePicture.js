import React from 'react';

import Card from 'react-bootstrap/Card';
import 'bootstrap/dist/css/bootstrap.min.css';

import { Columns } from '../styled/SinglePicture';

const SinglePicture = ({ picture, showModal }) => (
  <Columns>
    <Card
      className="picture"
      onClick={() => showModal()}
      border="dark"
      style={{ cursor: 'pointer', maxWidth: '200px' }}>
      <Card.Body
        style={{
          textAlign: 'center',
          minHeight: '180px',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center'
        }}>
        <Card.Img
          style={{ maxWidth: '100%', width: 'auto', height: '100%' }}
          className="picture__image"
          src={picture.previewURL}
        />
      </Card.Body>
    </Card>
  </Columns>
);

export default SinglePicture;
