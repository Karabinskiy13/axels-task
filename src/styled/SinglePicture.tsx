import styled from 'styled-components';

export const Columns = styled.div`
  display: flex;
  max-width: 300px;
  align-text: center;
  justify-content: center;
  align-items: center;
  min-height: 180px;
  border: 2px solid gray;
  border-radius: 10px;
  margin-left: auto;
  margin-right: auto;
`;
export const Modal = styled.button`
  border: none;
  background: none;
`;

export const Images = styled.img`
  width: 100%;
  height: 100%;
  max-width: 250px;
  max-height: 160px;
`;
