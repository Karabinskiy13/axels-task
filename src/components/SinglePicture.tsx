import React from 'react';

import Card from 'react-bootstrap/Card';

import { Columns } from '../styled/SinglePicture';
import { Image } from '../types';

interface Props {
  picture: Image;
  showModal: (status: boolean) => void;
  largeImageURL?: string;
  hits?: string;
}

const SinglePicture = ({ picture, showModal }: Props) => (
  <Columns>
    <Card
      className="picture"
      onClick={() => showModal(true)}
      border="dark"
      style={{ cursor: 'pointer', maxWidth: '200px' }}>
      <Card.Body
        style={{
          textAlign: 'center',
          minHeight: '180px',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center'
        }}>
        <Card.Img
          style={{ maxWidth: '100%', width: 'auto', height: '100%' }}
          className="picture__image"
          alt="picture__image"
          src={picture.previewURL}
        />
      </Card.Body>
    </Card>
  </Columns>
);

export default SinglePicture;
