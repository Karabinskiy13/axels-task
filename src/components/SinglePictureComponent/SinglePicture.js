import React from 'react';

import Card from 'react-bootstrap/Card';
import Col from 'react-bootstrap/Col';
import 'bootstrap/dist/css/bootstrap.min.css';

import { Picture } from '../../styled/SinglePicture';

const SinglePicture = ({ picture }) => {
  const { previewURL, tags } = picture;
  return (
    <Col>
      <Picture>
        <Card className="picture" border="dark" style={{ width: '18rem' }}>
          <Card.Body>
            <Card.Img
              style={{ maxHeight: '150px' }}
              className="picture__image"
              src={previewURL}></Card.Img>
            <Card.Header className="picture__title">{tags}</Card.Header>
          </Card.Body>
        </Card>
      </Picture>
    </Col>
  );
};

export default SinglePicture;
