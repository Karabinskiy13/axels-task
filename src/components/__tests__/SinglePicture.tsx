import React from 'react';
import { shallow } from 'enzyme';
import SinglePicture from '../SinglePicture';
import { Image } from '../../types';
import { jest } from '@jest/globals';

const singlePicture: Image = {
  id: 1,
  previewURL: '/image.png',
  tags: 'tags',
  largeImageURL: '/'
};

describe('<SinglePicture>', () => {
  it('Should render component', () => {
    const wrapper = shallow(<SinglePicture picture={singlePicture} showModal={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Calling callback on card click', () => {
    const showModalCb = jest.fn(() => true);
    const wrapper = shallow(<SinglePicture picture={singlePicture} showModal={showModalCb} />);
    const card = wrapper.find('.picture');
    card.simulate('click');
    expect(showModalCb).toBeCalledWith(true);
  });

  it('Should display props', () => {
    const wrapper = shallow(<SinglePicture picture={singlePicture} showModal={jest.fn()} />);
    expect(wrapper.find('.picture__image').prop('src')).toEqual(singlePicture.previewURL);
  });
});
