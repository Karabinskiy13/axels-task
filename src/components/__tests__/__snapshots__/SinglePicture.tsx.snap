// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePicture> Should render component 1`] = `
<Styled(Col)>
  <Card
    body={false}
    border="dark"
    className="picture"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "maxWidth": "200px",
      }
    }
  >
    <CardBody
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "minHeight": "180px",
          "textAlign": "center",
        }
      }
    >
      <CardImg
        className="picture__image"
        src="/image.png"
        style={
          Object {
            "height": "100%",
            "maxWidth": "100%",
            "width": "auto",
          }
        }
      />
    </CardBody>
  </Card>
</Styled(Col)>
`;
